include/zephyr/bluetooth/conn.h

```diff
@@ -233,7 +248,8 @@ void bt_conn_unref(struct bt_conn *conn);
  * @param func  Function to call for each connection.
  * @param data  Data to pass to the callback function.
  */
-void bt_conn_foreach(int type, void (*func)(struct bt_conn *conn, void *data),
+void bt_conn_foreach(bt_conn_type_t type,
+		     void (*func)(struct bt_conn *conn, void *data),
```

Comment:
I guess this is technically changing a stable API, right @jhedberg ? Should this go through the https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-incompatible-changes ?